### PR TITLE
Add metacoq dev packages

### PR DESCRIPTION
**Pull request by SkySkimmer**
ci-skip: coq-metacoq-erasure.dev coq-metacoq-pcuic.dev coq-metacoq-safechecker.dev coq-metacoq-template.dev coq-metacoq-translations.dev

### CODE REVIEW

File: extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
+opam-version: "2.0"
+maintainer: "matthieu.sozeau@inria.fr"
+homepage: "https://metacoq.github.io/metacoq"
+dev-repo: "git+https://github.com/MetaCoq/metacoq.git#coq-8.11"
```
Comment (silene) [inline extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam:4]: Bad copy-paste?

File: extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+]
+license: "MIT"
+build: [
+  ["sh" "./configure.sh"]
+  [make "-j" "%{jobs}%" "-C" "erasure"]
+]
+install: [
+  [make "-C" "erasure" "install"]
+]
+version: "dev"
```
Comment (palmskog) [inline extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam:24]:
```suggestion
```
Please avoid specifying `version` explicitly (it's given by the filename). Also in other files.

File: extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam (hunk 1 above)
Comment (SkySkimmer) [reply extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam]: that's what it says in the opam file in the repo

**Comment by SkySkimmer** [comment]
Looks like CI times out. Is there a way to make it install just the main metacoq package (testing the others through dependencies) or some other solution?

File: extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam (hunk 1 above)
Comment (palmskog) [reply extra-dev/packages/coq-metacoq-erasure/coq-metacoq-erasure.dev/opam]: Regardless of what it says in the repo opam file, please change to:
```
git+https://github.com/MetaCoq/metacoq.git#master
```

**Comment by palmskog** [comment]
@SkySkimmer there are basically two options:
1. Skip running CI for the "base" packages (which will be checked anyway) by following the instructions here: https://github.com/coq/opam-coq-archive#continuous-integration
2. Split this PR into a sequence of PRs with at most 2 packages each.

**Comment by SkySkimmer** [comment]
Looks like it has trouble running the tests with installed packages, so I dropped them.

**Comment by SkySkimmer** [comment]
CI passed, any objections to merge?

**Pull request merged**